### ISSUE TITLE
Another unexpected warning

### ISSUE THREAD

**Issue by kelloggm** [kelloggm/gradle-checker-framework-plugin#4]
In this repository: https://github.com/plume-lib/reflection-util in branch use-gradle-plugin, when I run

```
./gradlew build -Dorg.gradle.warning.mode=all
```

I get the warning message
```
The configuration :checkerFrameworkAnnotatedJDK was resolved without accessing the project in a safe manner.  This may happen when a configuration is resolved from a thread not managed by Gradle or from a different project.  See https://docs.gradle.org/5.3/userguide/troubleshooting_dependency_resolution.html#sub:configuration_resolution_constraints for more details. This behaviour has been deprecated and is scheduled to be removed in Gradle 6.0.
```

**Comment by kelloggm** [comment]
@mernst, I don't think this problem is actually *caused* by the CF gradle plugin - rather, it's exposed by it. This plugin adds additional configurations (`checkerFramework`, `checkerFrameworkAnnotatedJDK`, etc) to this project, which before only had one configuration. One of these (`checkerFrameworkAnnotatedJDK`) is being accessed somewhere it's not supposed to be. Where?

A bit of delta debugging led me to determine that commenting out these lines in the build file causes the warning to go away:
```groovy
configurations {
  requireJavadoc  // ensure that all Java elements have Javadoc documentation.
}
```
I'm not certain exactly how this is operating, which is why I'm punting this back to you. But a brief read of how `requireJavadoc` is defined suggests it is trying to resolve sources and apply a JavaDoc linter to them. I'm fairly certain it shouldn't be attempting to look at the `checkerFrameworkAnnotatedJDK` configuration, but I do not know whether there's anything that I, as the plugin author, can do to stop it!

**Comment by mernst** [comment]
Removing those lines didn't fix the problem for me.

I have created a smaller test case, which does not use `require-javadoc`.

You can find it in repository https://github.com/plume-lib/reflection-util in branch `use-gradle-plugin-without-requireJavadoc`.

Running
```./gradlew build -Dorg.gradle.warning.mode=all```
still gives the warning
```The configuration :checkerFrameworkAnnotatedJDK was resolved without accessing the project in a safe manner. ...```

Here is the full `build.gradle` file:

```groovy
// Checker Framework pluggable type-checking
buildscript {
  repositories {
    mavenLocal()
  }

  dependencies {
    classpath 'com.jaredsburrows:gradle-checker-framework-plugin:0.2.3-SNAPSHOT'
  }
}

plugins {
  id 'java-library'
}

apply plugin: 'com.jaredsburrows.checkerframework'

repositories {
  mavenCentral()
}

dependencies {
  testImplementation 'junit:junit:4.12'
}

sourceCompatibility = 1.8
targetCompatibility = 1.8

checkerFramework {
  checkers = [
    'org.checkerframework.checker.formatter.FormatterChecker',
  ]
}
```
